File: app/src/main/java/com/hhh/paws/database/dao/IdentificationDao.java
package com.hhh.paws.database.dao;

import com.hhh.paws.database.model.Identification;
import com.hhh.paws.util.UiState;

import java.util.function.Function;

import io.reactivex.rxjava3.core.Observable;
import kotlin.jvm.functions.Function1;

public interface IdentificationDao {
    Observable<Identification> getIdentification(String petName);
    void setIdentification(String petName, Identification identification);
}

File: app/src/main/java/com/hhh/paws/database/dao/IdentificationDao.kt
package com.hhh.paws.database.dao

import com.hhh.paws.database.model.Identification
import com.hhh.paws.util.UiState

interface IdentificationDao {
    suspend fun getIdentification(petName: String, result: (UiState<Identification>) -> Unit)
    suspend fun setIdentification(petName: String, identification: Identification, result: (UiState<String>) -> Unit)
}